Feature/s3 minor fixes

sametd opened the pull request:
Minor changes for s3:

- use_ssl is used external url
- separation of secure and use_ssl in minio
- removal of random_host

peshence commented on polytope_server/common/staging/s3_boto3_staging.py:
```diff
@@ -76,17 +74,9 @@ def __init__(self, config):
+        self.prefix = "https" if self.use_ssl else "http"
 
-        self._internal_url = f"{prefix}://{self.host}:{self.port}"
+        self._internal_url = f"http://{self.host}:{self.port}"
```
Is this prefix intentionally fixed now?

sametd replied:
This is for internal communication, and I do not believe we need to use https there at all. The use_ssl setting in the configuration should only affect download urls, not internal communication.

sametd commented:
Updated with complete rewrite of s3_staging.py, it works with latest version of minio-py. Also added depreciation warning to s3_staging.py . Furthermore, this implementation supports seaweedfs too.
